### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by VadimKadasiev**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,45 +3,168 @@
...
+    private Node<T> first;
+    private Node<T> last;
+
+    public int getListSize() {
+        return listSize;
+    }
+
+    public void setListSize(int listSize) {
+        this.listSize = listSize;
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/MyLinkedList.java:16]: useless methods, you can use the `size++` or `size--` where you need
```suggestion
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int listSize;
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/MyLinkedList.java:6]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,45 +3,168 @@
...
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return getListSize() == 0;
+    }
+
+    private Node<T> nodesIterate(int index) {
+        Node<T> testNode;
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/MyLinkedList.java:115]: do not use such naming, how to guess what it stores?
```suggestion
        Node<T> currentNode;
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,45 +3,168 @@
...
+        nodesIterate(index).item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldItem;
+        Node<T> testNode = nodesIterate(index);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/MyLinkedList.java:78]: the same

**Review by Ivan95kos — APPROVED** [review]
Great job, with minor comments

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 4 above)
Comment (VadimKadasiev) [reply src/main/java/core/basesyntax/MyLinkedList.java:78]: OK. Thank you!

**Pull request closed without merging**